src/main/java/io/leonis/subra/gui/field/SingleOrientationIndicator.java

```diff
@@ -0,0 +1,35 @@
+package io.leonis.subra.gui.field;
+
+import com.googlecode.lanterna.TerminalPosition;
+import io.leonis.algieba.geometry.CardinalDirection;
+import io.leonis.subra.game.data.Player;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+
+/**
+ * @author Jeroen de Jong
+ */
+@AllArgsConstructor
+public class SingleOrientationIndicator implements Function<TerminalPosition, Character> {
+  private final Character character;
+  private final List<TerminalPosition> positions;
+
+  public SingleOrientationIndicator(
+      final Set<Player> players,
+      final CardinalDirection cardinalDirection,
+      final TerminalPosition offset,
+      final Character character
+  ){
+    this(character, players.stream()
+        .filter(r -> CardinalDirection.from(r.getOrientation()).equals(cardinalDirection))
+        .map(robot -> offset.withRelative((int)robot.getX(), (int)robot.getY()))
```

Comment:
why not `Math.floor`? if you need rounding then typecasting will not work the same for negative numbers

Reply:
`Math.floor` returns the same type; i need to convert the double to an int. for now i don't care too much about whether it's floored or rounded unless there's a good reason to care :)